common/src/main/java/org/apache/falcon/util/DateUtil.java

```diff
@@ -99,4 +136,142 @@ public static Date now() {
+    public static Date parseDateOozieTZ(String s) throws ParseException {
+        s = s.trim();
+        ParsePosition pos = new ParsePosition(0);
+        Date d = getISO8601DateFormat(activeTimeZone, activeTimeMask).parse(s, pos);
+        if (d == null) {
+            throw new ParseException("Could not parse [" + s + "] using [" + activeTimeMask + "] mask",
+                    pos.getErrorIndex());
+        }
+        if (s.length() > pos.getIndex()) {
+            throw new ParseException("Correct datetime string is followed by invalid characters: " + s, pos.getIndex());
+        }
+        return d;
+    }
+
+    private static DateFormat getISO8601DateFormat(TimeZone tz, String mask) {
+        DateFormat dateFormat = new SimpleDateFormat(mask);
+        // Stricter parsing to prevent dates such as 2011-12-50T01:00Z (December 50th) from matching
+        dateFormat.setLenient(false);
+        dateFormat.setTimeZone(tz);
+        return dateFormat;
+    }
+
+    private static DateFormat getSpecificDateFormat(String format) {
+        DateFormat dateFormat = new SimpleDateFormat(format);
+        dateFormat.setTimeZone(activeTimeZone);
+        return dateFormat;
+    }
+
+    /**
+     * Formats a {@link java.util.Date} as a string using the specified format mask.
+     * <p/>
+     * The format mask must be a {@link java.text.SimpleDateFormat} valid format mask.
+     *
+     * @param d {@link java.util.Date} to format.
+     * @return the string for the given date using the specified format mask,
+     * <code>NULL</code> if the {@link java.util.Date} instance was <code>NULL</code>
+     */
+    public static String formatDateCustom(Date d, String format) {
+        return (d != null) ? getSpecificDateFormat(format).format(d) : "NULL";
+    }
+
+    /**
+     * Formats a {@link java.util.Date} as a string in ISO8601 format using Oozie processing timezone.
+     *
+     * @param d {@link java.util.Date} to format.
+     * @return the ISO8601 string for the given date, <code>NULL</code> if the {@link java.util.Date} instance was
+     * <code>NULL</code>
+     */
+    public static String formatDateOozieTZ(Date d) {
+        return (d != null) ? getISO8601DateFormat(activeTimeZone, activeTimeMask).format(d) : "NULL";
+    }
+
+    /**
+     * Returns the {@link java.util.TimeZone} for the given timezone ID.
+     *
+     * @param tzId timezone ID.
+     * @return  the {@link java.util.TimeZone} for the given timezone ID.
+     */
+    public static TimeZone getTimeZone(String tzId) {
+        if (tzId == null) {
+            throw new IllegalArgumentException("Timezone cannot be null");
+        }
+        tzId = handleGMTOffsetTZNames(tzId);    // account for GMT-####
+        TimeZone tz = TimeZone.getTimeZone(tzId);
+        // If these are not equal, it means that the tzId is not valid (invalid tzId's return GMT)
+        if (!tz.getID().equals(tzId)) {
+            throw new IllegalArgumentException("Invalid TimeZone: " + tzId);
```

Comment:
will not this validation happen at the time of entity submit itself. Will this not be required to check again during schedule?